component/src/main/java/org/wso2/extension/siddhi/io/ibmmq/sink/IBMMQSink.java

```diff
@@ -0,0 +1,244 @@
+                    validateAndGetOption(IBMMQConstants.DESTINATION_NAME).getValue() +
+                    ", " + e.getMessage(), e);
+        }
+
+    }
+
+    @Override
+    public void publish(Object payload, DynamicOptions transportOptions) throws ConnectionUnavailableException {
+        try {
+            if (payload instanceof String) {
+                Message message = session.createTextMessage(payload.toString());
+                messageSender.send(message);
+
+            } else if (payload instanceof Map) {
+                MapMessage mapMessage = session.createMapMessage();
+                ((Map) payload).forEach((key, value) -> {
+                    try {
+                        mapMessage.setString((String) key, (String) value);
+                    } catch (JMSException e) {
+                        throw new IBMMQSinkAdaptorRuntimeException("Exception occurred while publishing payload: " +
+                                "key - '" + key + "', value - '" + value + "' from stream: '"
+                                + outputStreamDefinition.getId() + "'. ", e);
+                    }
+                });
+                messageSender.send(mapMessage);
+            } else if (payload instanceof ByteBuffer) {
+                byte[] data = ((ByteBuffer) payload).array();
+                BytesMessage bytesMessage = session.createBytesMessage();
+                bytesMessage.writeBytes(data);
+                messageSender.send(bytesMessage);
+            }
+            messageSender.close();
+        } catch (JMSException e) {
+            throw new IBMMQSinkAdaptorRuntimeException("Exception occurred while publishing payload: " +
+                    payload.toString() + " , ", e);
+        }
+    }
+
+    @Override
+    public void connect() throws ConnectionUnavailableException {
+        try {
+            if (isSecured) {
+                connection = (QueueConnection) connectionFactory.createConnection(userName, password);
+            } else {
+                connection = (QueueConnection) connectionFactory.createConnection();
+            }
+            session = (QueueSession) connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+            queue = session.createQueue(optionHolder.validateAndGetOption(IBMMQConstants.DESTINATION_NAME)
+                    .getValue());
+            consumer = session.createConsumer(queue);
+            messageSender = session.createSender(queue);
+
+        } catch (JMSException e) {
+            throw new ConnectionUnavailableException("Exception occurred while connecting to the IBM MQ for queue: '"
+                    + queueName + "' in siddhi app: '" + siddhiAppContext.getName() + "'. ", e);
+        }
+    }
+
+    @Override
+    public void disconnect() {
+        try {
+            if (Objects.nonNull(connection)) {
+                connection.close();
+            }
+            if (Objects.nonNull(consumer)) {
+                consumer.close();
```

Comment:
if exception was thrown in connection.close(); then consumer.close(); will not be called